DedicatedServer/Scripts/Libs/Rx/ModeLibs/Common/Hitbox.script.txt

```diff
@@ -0,0 +1,159 @@
+#Const Version					"2021-12-22"
+#Const ScriptName				"Hitbox.Script.txt"
+
+#Include "MathLib" as ML
+
+#Const C_UpOffset 		0.44
+#Const C_FrontOffset	1.3
+#Const C_BackOffset 	0.75
+
+#Const C_CarEllipsoid     	<1.1, 0.5, 1.1> //xyz "radius"
+#Const C_CarEllipsoidOffset <-0.3, 0.575, -0.3>
+#Const C_FlagEllipsoid			<1.5,3.,1.5>
+
+/**
+ * recompute value below if you changed any value above
+ */
+#Const C_MaxDist 23.04
+#Const C_MinDist 1
+#Const C_MaxDistFlag 16.6
+#Const C_MinDistFlag 5.29
+
+/**
+* double vec3 struct to be used either as :
+* a segment (point - point)
+* an ellipsoid (point - xyz "radius")
+*/
+#Struct Struct_Vec3Vec3 {
+	Vec3 A;
+	Vec3 B;
+}
+/**
+* fast Sphere distance function
+* return the squared distance to avoid a sqrt
+*/
+Real DistanceSquared(Vec3 _A, Vec3 _B){
+	declare Real X = (_A.X - _B.X);
+	declare Real Y = (_A.Y - _B.Y);
+	declare Real Z = (_A.Z - _B.Z);
+	return X * X + Y * Y + Z * Z;
+}
+
+/**
+* fast Ellipsoid distance function
+* return the squared distance to avoid a sqrt
+*/
+Real WeightedDistanceSquared(Vec3 _A, Vec3 _B, Vec3 _V, Vec3 _W) {
+	declare Real X = (_A.X - _B.X) / (_V.X + _W.X);
+	declare Real Y = (_A.Y - _B.Y) / (_V.Y + _W.Y);
+	declare Real Z = (_A.Z - _B.Z) / (_V.Z + _W.Z);
+	return X * X + Y * Y + Z * Z;
+}
+
+/**
+* get the ellipsoid of the collider of the player
+* according to the player orientation
+*/
+Vec3 PlayerEllipsoid(CSmPlayer Player){
+	declare Real Sideness = 1-ML::Abs(Player.Upwardness);
```

Comment:
Isn't Upwardness in range [0.0, 1,0]? Pretty sure I checked that today. That would make `ML::Abs()` unnecessary. Also, wouldn't it the be "Downness"? ^^

Reply:
Good catch this should be `1-ML::Abs(Player.Upwardness-0.5)*2`, I thought it was in range [-1.0,1.0] initially in my C# re-implementation.